src/function/scalar/string/hex.cpp

```diff
@@ -0,0 +1,217 @@
+
+		bool seen_non_zero = false;
+		for (idx_t offset = sizeof(INPUT_TYPE) * 8; offset >= 8; offset -= 8) {
+			uint8_t byte = (uint8_t)(input >> (offset - 8));
+			// offset == 8: do not continue. The result has at least one byte
+			if (byte == 0 && !seen_non_zero && offset > 8) {
+				continue;
+			}
+			seen_non_zero = true;
+			memcpy(ptr, &StringUtil::BYTE_TO_HEXS[static_cast<idx_t>(byte) * 2], 2);
+			ptr += 2;
+			buffer_size += 2;
+		}
+
+		// Allocate empty space
+		auto target = StringVector::EmptyString(result, buffer_size);
+		auto output = target.GetDataWriteable();
+		memcpy(output, buffer, buffer_size);
+
+		target.Finalize();
+		return target;
+	}
+};
+
+struct HexHugeIntOperator {
+	template <class INPUT_TYPE, class RESULT_TYPE>
+	static RESULT_TYPE Operation(INPUT_TYPE input, Vector &result) {
+		char buffer[sizeof(INPUT_TYPE) * 2];
+		char *ptr = buffer;
+		idx_t buffer_size = 0;
+
+		uint64_t lower = input.lower;
+		int64_t upper = input.upper;
+
+		bool seen_non_zero = false;
+		for (idx_t offset = 64; offset >= 8; offset -= 8) {
+			uint8_t byte = (uint8_t)(upper >> (offset - 8));
+
+			if (byte == 0 && !seen_non_zero) {
+				continue;
+			}
+			seen_non_zero = true;
+			memcpy(ptr, &StringUtil::BYTE_TO_HEXS[static_cast<idx_t>(byte) * 2], 2);
+			ptr += 2;
+			buffer_size += 2;
+		}
+
+		for (idx_t offset = 64; offset >= 8; offset -= 8) {
+			uint8_t byte = (uint8_t)(lower >> (offset - 8));
+
+			// at least one byte space
+			if (byte == 0 && !seen_non_zero && offset > 8) {
+				continue;
+			}
+			seen_non_zero = true;
+			memcpy(ptr, &StringUtil::BYTE_TO_HEXS[static_cast<idx_t>(byte) * 2], 2);
+			ptr += 2;
+			buffer_size += 2;
+		}
+
+		// Allocate empty space
+		auto target = StringVector::EmptyString(result, buffer_size);
+		auto output = target.GetDataWriteable();
+		memcpy(output, buffer, buffer_size);
+
+		target.Finalize();
+		return target;
+	}
+};
+
+static void ToHexFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto &input = args.data[0];
+	idx_t count = args.size();
+
+	switch (input.GetType().InternalType()) {
+	case PhysicalType::VARCHAR:
+		UnaryExecutor::ExecuteString<string_t, string_t, HexStrOperator>(input, result, count);
+		break;
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		UnaryExecutor::ExecuteString<int8_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT16:
+		UnaryExecutor::ExecuteString<int16_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT32:
+		UnaryExecutor::ExecuteString<int32_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT64:
+		UnaryExecutor::ExecuteString<int64_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT128:
+		UnaryExecutor::ExecuteString<hugeint_t, string_t, HexHugeIntOperator>(input, result, count);
+		break;
+	case PhysicalType::UINT8:
```

Comment:
I think it is probably only necessary to add `UINT64`, `INT64` `HUGEINT` and `VARCHAR`. The other data types will be supported using automatic casts.